### PR TITLE
Generate developer docs for Python API

### LINKED ISSUE #492: Generate API documentation using `mkdocstrings`
Use https://github.com/mkdocstrings/mkdocstrings to create docs for the new's KPOps Python API

### PR DESCRIPTION
**Pull request by sujuka99**
Closes #492

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -37,6 +37,23 @@ plugins:
...
+          paths: [..]
+          options:
+            docstring_style: sphinx
+            show_if_no_docstring: true
+            line_length: 60
+            separate_signature: true
+            show_signature_annotations: true
+            docstring_section_style: spacy
+            show_root_heading: true
+            # The options below are not yet publicly available,
```
Comment (raminqaf) [inline docs/mkdocs.yml:52]: is there a link to this?

File: docs/docs/developer/api.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+<!-- dprint-ignore-start -->
+
+::: kpops.api
+
+<!-- dprint-ignore-end -->
+
+<!-- dprint-ignore-start -->
+
+::: kpops.pipeline.Pipeline
```
Comment (raminqaf) [inline docs/docs/developer/api.md:11]: I don't really the `Pipeline` class should be in the public API docs.

**Comment by raminqaf** [comment]
Is there a way to omit `setup_handler` and `create_pipeline` from the API docs?

Besides, it would be nice to have some PyDocs under each method so it is reflected in the docs

image

File: docs/mkdocs.yml (hunk 1 above)
Comment (sujuka99) [reply docs/mkdocs.yml:52]: Added links

File: docs/docs/developer/api.md (hunk 2 above)
Comment (sujuka99) [reply docs/docs/developer/api.md]: Removed

Comment (disrupted) [reply docs/docs/developer/api.md]: yes, `Pipeline` can be made public imo

**Comment by disrupted** [comment]
> Is there a way to omit `setup_handler` and `create_pipeline` from the API docs?
> 
> Besides, it would be nice to have some PyDocs under each method so it is reflected in the docs
> 
> image

rather than manually hiding those, I would prefer prefixing them with `_` to make them private and then only exporting the public ones.

**Review by raminqaf — APPROVED** [review]

**Pull request merged**